openedx_events/content_authoring/data.py

```diff
@@ -0,0 +1,117 @@
+"""
+Data attributes for events within the architecture subdomain ``content_authoring``.
+
+These attributes follow the form of attr objects specified in OEP-49 data
+pattern.
+"""
+from datetime import datetime
+
+import attr
+from opaque_keys.edx.keys import CourseKey
+
+
+@attr.s(frozen=True)
+class MediaData:
+    """
+    Data for a media object (image or video).
+
+    Arguments:
+        uri (str): media object filename
+        absolute_uri (str): fully qualified media object location
+    """
+
+    uri = attr.ib(type=str, default=None)
+    uri_absolute = attr.ib(type=str, default=None)
+
+
+@attr.s(frozen=True)
+class ImageData:
+    """
+    Data for a display image.
+
+    Arguments:
+        raw (str): uri of the raw image (required)
+        small (str): uri of a small version of the image (optional)
+        large (str): uri of a large version of the image (optional)
+    """
+
+    raw = attr.ib(type=str)
+    small = attr.ib(type=str, default=None)
+    large = attr.ib(type=str, default=None)
+
+
+@attr.s(frozen=True)
+class CourseMediaData:
+    """
+    Data describing media for the course catalog display.
+
+    Arguments:
+        banner_image (MediaData): top banner image (optional)
+        course_image (MediaData): catalog image (optional)
+        course_video (MediaData): catalog video (optional)
+        image (ImageData): course card image
+    """
+
+    banner_image = attr.ib(type=MediaData, default=None)
+    course_image = attr.ib(type=MediaData, default=None)
+    course_video = attr.ib(type=MediaData, default=None)
+    image = attr.ib(type=ImageData, default=None)
```

Comment:
We could update ImageData to be
`
uri_raw
uri_absolute_raw
uri_small
uri_absolute_small
uri_large
uri_absolute_large
`
it will require a bit of fudging to get all the data in the correct fields but it means we can reuse it.

Reply:
In response to https://github.com/openedx/openedx-events/pull/81#issuecomment-1192730301:

> 1. Copy the existing fields exactly as they are from the LMS response and just acknowledge that they are weird and inconsistent. We could still fix the field names to be a little clearer.

I was wondering if we could start with a slightly edited version of option (1), and skip the current MediaData class. Let's provide the fields directly in the higher level Data class, and always provide exactly one str of either url or path. 

Notes: 
* MultiSizeImageData (or whatever you want to call it), could still be a separate class if you want it.
* If you have both url and path, just supply the url and the path can be calculated by the consumer. (This is a bit of option 3.)
* If you just have path, we'll learn later if it can easily be changed to a url and we can update this definition later.

Of course this all depends on your research of where this data is coming from in the first place. If it doesn't come from Studio editing, it doesn't really belong as part of this event.

Reply:
Resolved by dropping media stuff